File: SE/20230213_기사단원의무기.java
import java.util.*;

class Solution {
    public int solution(int number, int limit, int power) {
        int answer = 0;
        int[] soldier = new int[number];
        for (int i=0; i<soldier.length; i++ ) {
            soldier[i] = i+1;
        }
        
        int[] m = new int[number];
        for (int i =0; i<soldier.length; i++) {
            for (int j =1; j*j<=soldier[i]; j++) {
                if (j*j == soldier[i]) m[i]++;
                else if(soldier[i] % j == 0) {
                    m[i] += 2;
                }
            }
            //System.out.println(m[i]);
        }
        
        for(int i : m) {
            if (i > limit) {
                answer+=power;
            } else {
                answer+=i;    
            }
            //System.out.println(answer);
        }
        return answer;
    }
}

File: tottale/20230213-기사단원의무기.swift
import Foundation

func solution(_ number:Int, _ limit:Int, _ power:Int) -> Int {
    var steels = 0
    for solNum in 1...number {
        var solPower = 0
        var i = 1
        while i*i <= solNum {
            if i*i == solNum {
                solPower += 1
            } else if solNum%i == 0 {
                solPower += 2
            }
            i += 1
        }
        if solPower > limit {
            steels += power
        } else {
            steels += solPower
        }
    }
    return steels
}
